src/lib/gundb/gundb-extend.js

```diff
@@ -91,6 +91,34 @@ assign(chain, {
 
     return (callback || identity)(decryptedData)
   },
+
+  /**
+   * Returns the decrypted value
+   * @returns {Promise<any>}
+   */
+  // eslint-disable-next-line require-await
+  async onDecrypt(callback = null) {
+    let _resolve
+    let decryptedData = null
+    const resolve = value => over([callback, _resolve].filter(isFunction))(value)
+    const promise = new Promise(resolve => (_resolve = resolve))
```

Comment:
same comment as above. revert this change and move event.off to getOutbox just like i do in getcounterparty

Reply:
event.off is **inside** GUN.on, exactly like in the counterparty
the logic of this method kept untouched. 
it just additionally returns an promise resolves when we getting first data and callback invokes